src/app/core/store/collections/collections.reducer.ts

```diff
@@ -55,28 +58,39 @@ export function collectionsReducer(state: CollectionsState = initialCollectionsS
   }
 }
 
+function onCreateAttributeSuccess(state: CollectionsState, action: CollectionsAction.CreateAttributeSuccess): CollectionsState {
+  let attributes = state.entities[action.payload.collectionId].attributes.slice();
+  attributes.push(action.payload.attribute);
+  return collectionsAdapter.updateOne({id: action.payload.collectionId, changes: {attributes: attributes}}, state);
+}
+
 function onChangeAttributeSuccess(state: CollectionsState, action: CollectionsAction.ChangeAttributeSuccess): CollectionsState {
   let attributes = state.entities[action.payload.collectionId].attributes.slice();
   const index = attributes.findIndex(attr => attr.id === action.payload.attributeId);
+  const oldAttributeCopy = index >= 0 ? {...attributes[index]} : null;
   if (index >= 0) {
     attributes.splice(index, 1, action.payload.attribute);
   } else {
     attributes.push(action.payload.attribute); // TODO preserve order
   }
 
-  if (action.payload.attributeId !== action.payload.attribute.id) {
-    attributes = renameChildAttributes(attributes, action.payload.attributeId, action.payload.attribute.id);
+  if(!isNullOrUndefined(oldAttributeCopy)){
+
+  }
```

Comment:
What should happen in this case? I guess nothing but we need to add "null check" to the condition a few lines below.